cmd/skopeo/sync.go

```diff
@@ -74,6 +75,25 @@ type registrySyncConfig struct {
 // sourceConfig contains all registries information read from the source YAML file
 type sourceConfig map[string]registrySyncConfig
 
+// syncConfig contains all registries information read from the source YAML file for source yaml2
+type syncConfig map[string]registrySyncConfigV2
+
+// registrySyncConfigV2 contains all information read from the yaml2 sync config for a registry
+type registrySyncConfigV2 struct {
+	Repos       map[string]repoSyncConfig
+	Credentials types.DockerAuthConfig // Username and password used to authenticate with the registry
```

Comment:
Are credentials really a per-`SyncConfig` thing?

Conceptually we need both source and destination credentials. How does that work here?

(I _guess_ credentials should be a separate top-level map but I’m not at all sure about that.)